cmd/sonictool/chain/export_events.go

```diff
@@ -24,15 +26,22 @@ var (
 // always print out progress. This avoids the user wondering what's going on.
 const statsReportLimit = 8 * time.Second
 
-func ExportEvents(w io.Writer, dataDir string, from, to idx.Epoch) (err error) {
+func ExportEvents(ctx *cli.Context, w io.Writer, dataDir string, from, to idx.Epoch) (err error) {
```

Comment:
nit: passing the full context to the `ExportEvents` and `ImportGenesisStore` is maybe over-sized and it moves the interpretation of flags into the implementations of those functions. I would instead create a storage-config struct including the cache sizes and maybe the data-dir. Wdyt?

Reply:
I dont like adding another struct for config - there is quite a mess in that. How about I use the already created struct `GossipDbParameters` and just pass in the Cache sizes and dir and fill the rest  of the struct later?

Reply:
generating `GossipDbParameters` from the command line parsing could be an option. But if this turns cumbersome, please use a new struct with two fields for
```
	LiveDbCache:   ctx.Int64(flags.LiveDbCacheFlag.Name),
	ArchiveCache:  ctx.Int64(flags.ArchiveCacheFlag.Name),
```